controller.go

```diff
@@ -284,13 +284,17 @@ func syncPV(pv *PV) {
 			// NOTE: releasePV may either release the PV back into the pool or
 			// recycle it or do nothing (retain)
 
-			// HOWTO RELEASE A PV
-			pv.Status.Phase = Released
-			if err := CommitPVStatus(pv); err != nil {
-				// Status was not saved; we will fall back into the same
-				// condition in the next call to this method
-				return
+			// Make sure we don't overwrite previous failure of
+			// recycler/deleter.
+			if pv.Status.Phase != Failed {
```

Comment:
> PV.Status.Phase is explicitly set to Failed here: https://github.com/pmorie/pv-haxxz/blob/master/controller.go#L331

I see.. so that one should be ok because the volume will try to recycle again and continue to not have a recycling plugin and continue to be marked as failed.

> It's not in our pseudocode, but I set it to Failed also every time that recycling fails, e.g. steps 2.5 and 4 (pod fails) in the recycling process at lines https://github.com/pmorie/pv-haxxz/blob/master/controller.go#L318

These are ok too I think because nothing about the PV was changed so it should continue to fall into the same path and be marked as failed.

I am just trying to avoid depending on phase when making decisions as it was one of our design goals.

It would be worth adding setting status to failed into the pseudo code just to call it out explicitly.